pubsub/topic.go

```diff
@@ -534,6 +644,48 @@ func (i *IngestionDataSourceSettings) toProto() *pb.IngestionDataSourceSettings
 				},
 			}
 		}
+		if cs, ok := out.(*IngestionDataSourceCloudStorage); ok {
+			switch format := cs.InputFormat.(type) {
+			case *IngestionDataSourceCloudStorageTextFormat:
```

Comment:
Could a similar method to what is used in `protoToIngestionDataSourceSettings` be used for this function, so that you create an InputFormat proto object within the switch statement and then declare the full proto once at the end? There may be some proto limitations that I'm unaware of that make this difficult/impossible.

Reply:
Yeah this looks messy/redundant because there is no single "InputFormat" proto object that I can assign to, since it's not a public interface but [private](https://github.com/googleapis/google-cloud-go/blob/3377a3c709b66e14ff02e564014fedc231f5d25a/pubsub/apiv1/pubsubpb/pubsub.pb.go#L4450).

I could define a custom interface, but given that we're moving to GAPIC admin clients anyway, I think this is an ok compromise.